src/NServiceBus.Core/Diagnostics/TracingHelper.cs

```diff
@@ -0,0 +1,36 @@
+﻿namespace NServiceBus;
+
+using System;
+using System.Diagnostics;
+using System.Threading.Tasks;
+
+class TracingHelper
+{
+#pragma warning disable PS0013 // A Func used as a method parameter with a Task, ValueTask, or ValueTask<T> return type argument should have at least one CancellationToken parameter type argument unless it has a parameter type argument implementing ICancellableContext
+#pragma warning disable PS0018 // A task-returning method should have a CancellationToken parameter unless it has a parameter implementing ICancellableContext
+    public static async Task TryTraceInvocation<T1>(Activity activity, Func<T1, Task> method, T1 args)
```

Comment:
I believe that every invocation of this has `T1` being a tuple of some kind of invocation/pipeline and a context. This could be written as

```csharp
public static async Task TryTraceInvocation<TInvocation, TContext>(Activity activity, Func<TInvocation, TContext, Task> method, TInvocation invocation, TContext context)
```

The we wouldn't have to wrap and unwrap the tuple every time.

Reply:
yeah, this method could probably be further optimized, we can also ping Daniel if we have something mergeable for a review